src/vcpu.rs

```diff
@@ -309,8 +317,8 @@ impl RISCVVCpu {
                     counter_mask as usize,
                     stop_flags as usize,
                 );
-                self.set_gpr(GprIndex::A0, sbi_ret.error);
-                self.set_gpr(GprIndex::A1, sbi_ret.value);
+                self.set_gpr(GprIndex::A0 as usize, sbi_ret.error);
+                self.set_gpr(GprIndex::A1 as usize, sbi_ret.value);
```

Comment:
so many redundant `as usize`

Reply:
Do I need to add a function, such as set_gpr_from_index, to change the original set_gpr usize parameter to the GprIndex type?

Reply:
```Rust
impl GeneralPurposeRegisters {
    pub fn set_reg(&mut self, reg_index: GprIndex, val: usize) {}
}

impl AxArchVCpu for RISCVVCpu {
    fn set_gpr(&mut self, index: usize, val: usize) {
        self.GeneralPurposeRegisters.set_gpr(GprIndex::from(index), val);
    }
}
```

something like this may work